scalyr_agent/agent_main.py

```diff
@@ -689,9 +708,126 @@ def __handle_terminate(self):
+
+            if not os.path.exists(pidfile):
+                return {"error": "No PID file."}
+
+            try:
+                with open(pidfile, "r") as fp:
+                    pidfile_content = fp.read().strip()
+            except OSError as e:
+                return {"error": "Error during PID file read. Error: {}".format(e)}
+
+            try:
+                psutil_process = psutil.Process(pid=int(pidfile_content))
+            except psutil.Error as e:
+                return {"error": "Can't create psutil process. Error: {}".format(e)}
+
+            try:
+                with psutil_process.oneshot():
+                    agent_process_stats["pid"] = psutil_process.pid
+                    agent_process_stats["cpu_percent"] = psutil_process.cpu_percent()
+                    agent_process_stats["memory_rss"] = psutil_process.memory_info().rss
+            except Exception as e:
+                agent_process_stats["error"] = str(e)
+
+            return agent_process_stats
+
+        # Get information about machine's current state.
+        def get_machine_stats():
+            machine_stats = {}
+            try:
+                machine_stats["cpu_percent"] = psutil.cpu_percent()
+                machine_stats["cpu_count"] = psutil.cpu_count()
+                machine_stats["getloadavg"] = psutil.getloadavg()
+                machine_stats[
+                    "virtual_memory_percent"
+                ] = psutil.virtual_memory().percent
+            except Exception as e:
+                machine_stats["error"] = str(e)
+
+            return machine_stats
+
+        # Get list of related processes'
+        def find_agent_processes():
+            agent_processes = []
+            try:
+                ps_output = subprocess.check_output(
+                    [
+                        "ps",
+                        "aux",
+                    ]
+                ).decode()
+            except Exception as e:
+                return {"error": "Can't get list of processes. Error: {}".format(e)}
+
+            for line in ps_output.splitlines():
+                if "scalyr-agent" in line or "python" in line or "agent_main" in line:
+                    agent_processes.append(line)
+
+            return agent_processes
+
+        # Get status file stats and content.
+        def get_status_files_info():
+            result = {}
+            try:
+                ls_output = subprocess.check_output(
+                    ["ls", "-la", data_directory]
+                ).decode()
+                result["data_root_file_stats"] = ls_output.splitlines()
+            except Exception as e:
+                result["data_root_file_stats"] = six.text_type(e)
+
+            try:
+                ls_output = subprocess.check_output(
+                    [
+                        "ls",
+                        "-la",
+                        os.path.join(tempfile.gettempdir(), TEMP_STATUS_FILES_DIR_NAME),
```

Comment:
Did you forgot to update / revert this place back to using agent data dir?

Reply:
ah, yes, removed. Thanks for noticing that.